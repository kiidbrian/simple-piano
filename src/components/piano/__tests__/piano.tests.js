import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { render } from '@testing-library/react';
import { mount } from 'enzyme';
import Brand from '../brand';
import PianoScale from '../scale';
import PianoKey from '../key';

// Configure Enzyme
Enzyme.configure({ adapter: new Adapter() })

describe('<Brand/>', () => {
    it('renders the Brand', () => {
        const container = render(<Brand />)
        expect(container.firstChild).toMatchSnapshot()
    })

    it('renders the piano brand', () => {
        const { getByText } = render(<Brand />);
        const linkElement = getByText(/Roland/i);
        expect(linkElement).toBeInTheDocument();
    });
})

describe('<PianoScale/>', () => {
    it('renders the PianoScale', () => {
        const container = render(<PianoScale />)
        expect(container.firstChild).toMatchSnapshot()
    })
})

describe('<PianoKey/>', () => {
    const mockFn = jest.fn();

    const props = {
        note: 'A', 
        isFlat: false, 
        isHalfStep: false, 
        octave: 2, 
        onStart: mockFn, 
        onStop: mockFn,
        selectedNote: 'A'
    }

    it('renders the PianoKey', () => {
        const container = mount(<PianoKey {...props} />)
        container.find('li').simulate('click')
        expect(mockFn).toHaveBeenCalled()
    })
})

