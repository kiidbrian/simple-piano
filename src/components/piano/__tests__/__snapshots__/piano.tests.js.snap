// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Brand/> renders the Brand 1`] = `undefined`;

exports[`<Brand/> renders the Brand 2`] = `undefined`;

exports[`<Pi/> renders the Brand 1`] = `undefined`;

exports[`<PianoScale/> renders the  1`] = `undefined`;

exports[`<PianoScale/> renders the PianoScale 1`] = `undefined`;
